lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -342,7 +342,7 @@ def gem_version_compare(first, second)
                      curl --cacert $SSL_CA_CERT https://smart-proxy.example.com:8443"
           end
           def foreman_server_ca_cert
-            if File.exist?(Setting[:ssl_ca_file])
+            if File.exist?(Setting[:ssl_ca_file].to_s)
```

Comment:
I don't like this and think the following is much cleaner:
```suggestion
            if !Setting[:ssl_ca_file].blank? && File.exist?(Setting[:ssl_ca_file])
```

My reasoning is that `File.exist?` calls `stat` which is a syscall. If you can avoid those, it's always good while throwing garbage at the Kernel isn't.

Btw, I'm not entirely sure why we have a `File.exist?` check at all. It doesn't give you a guarantee that `File.read` will pass. It may be unreadable for example. Could also be a directory.

Reply:
Changed code to:
```
def foreman_server_ca_cert
  File.read(Setting[:ssl_ca_file])
rescue
  msg = N_("SSL CA file not found, check the 'SSL CA file' in Settings > Authentication")
  raise Foreman::Exception.new(msg)
end
```
This should cover all the problems

Reply:
Wouldn't ir confuse users if the proper permissions are missing even though the file exists?

Reply:
> Wouldn't ir confuse users if the proper permissions are missing even though the file exists?

Good point, changed to:

```
echo "ERROR: standard_error";
echo "ERF42-8752 [Foreman::Exception]: <File.read error>, check the 'SSL CA file' in Settings > Authentication";
exit 1
```
```